lib/engine/game/g_18_zoo/step/choose_ability_on_or.rb

```diff
@@ -0,0 +1,261 @@
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_sugar?(corporation)
+        #   @game.a_spoonful_of_sugar.owner == corporation &&
+        #     corporation.trains.any? { |train| !%w[2J 4J].include? train.name } &&
+        #     corporation.all_abilities.none? { |a| a.type == :increase_distance_for_train }
+        # end
+
+        def choices_for_zoo_ticket(company)
+          corporation = @game.current_entity
+          name = company.owner.name
+          [*1..company.value].map do |value|
+            [
+              { type: :sell, price: value, corporation: corporation.id },
+              "Sells for #{@game.format_currency(value)}; #{@game.format_currency(company.value - value)} to #{name}",
+            ]
+          end.to_h
+        end
+
+        # TODO: add logic
+        # def choices_for_moles?(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic
+        # def choices_for_ancient_maps?(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic
+        # def choices_for_hole?(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic
+        # def choices_for_on_diet?(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic
+        # def choices_for_is_mine(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic
+        # def choices_for_work_in_progress?(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic
+        # def choices_for_two_barrels?(_corporation)
+        #   { { type: :two_barrels } => "#{@game.two_barrels.all_abilities[0].count == 2 ? 'First' : 'Last' } barrel" }
+        # end
+
+        # TODO: add logic
+        # def choices_for_bandage?(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic
+        # def choices_for_wings?(_corporation)
+        #   {}
+        # end
+
+        # TODO: add logic for 'Bandage'
+        # Not available for 2J / 4J, or if train has the 'Bandage'
+        # def choices_for_sugar(corporation)
+        #   corporation.trains
+        #              .reject { |train| %w[2J 4J].include? train.name }
+        #              .map { |train| [{ type: :sugar, train_id: train.id }, train.name] }
+        #              .to_h
+        # end
+
+        def process_choose_zoo_ticket(action)
+          company = action.entity
+          player = company.owner
+          corporation = @game.corporation_by_id(action.choice['corporation'])
+          price_for_corporation = action.choice['price']
+          price_for_player = company.value - price_for_corporation
+
+          @log << "-- #{corporation.name} sells #{company.name} (owned by #{player.name}) "\
+            "for #{@game.format_currency(company.value)} --"
+
+          @game.bank.spend(price_for_player, player) if price_for_player.positive?
+          @log << "#{player.name} earns #{@game.format_currency(price_for_player)}" if price_for_player.positive?
```

Comment:
encapsulate in one if statement instead of checking if price_for_player is positive twice

Reply:
done in 2edf5ad23